presto-hive/src/main/java/com/facebook/presto/hive/rcfile/RcFilePageSource.java

```diff
@@ -349,6 +349,20 @@ public String toString()
                 .toString();
     }
 
+    @Override
+    public long getDeltaMemory()
+    {
+        long deltaMemory = 0;
+        try {
+            deltaMemory = recordReader.getCurrentKeyBufferObj().getSize();
+        }
+        catch (IOException e) {
+            closeWithSuppression(e);
```

Comment:
Why are we closing the object and failing the read just because the size check failed?